src/physics/physicsmodel.cxx

```diff
@@ -34,7 +34,8 @@
 
 #include <bout/mesh.hxx>
 
-PhysicsModel::PhysicsModel() : modelMonitor(this) {
+PhysicsModel::PhysicsModel()
+    : mesh(bout::globals::mesh), dump(bout::globals::dump), modelMonitor(this) {
```

Comment:
Should we provide a constructor that allows passing a `mesh` or `dump` that's different from the global one? E.g. with `nullptr` default values
```
PhysicsModel::PhysicsModel(Mesh* m, DataFile* d)
    : mesh(m == nullptr ? bout::globals::mesh : m),
      dump(d == nullptr ? *bout::globals::dump : *d),
      modelMonitor(this) {
```

Reply:
I started to do that, but we'd need constructors on the actual user model for this to be useful:

```cpp
class MyModel : public PhysicsModel {
public:
  MyModel(Mesh* mesh, DataFile* dump) : PhysicsModel(mesh, dump) {}
...
};

// In BOUTMAIN():

// This needs to call MyModel ctor
auto model = bout::utils::make_unique<ModelClass>(mesh, dump);         \
```

We can add the constructor as you suggest on `PhysicsModel` now, we just can't call it in `BOUTMAIN` yet!

Reply:
Ah, I see. Maybe a change for another day then, especially since adding more arguments to the constructor won't break backward compatibility so no need to get it in early.